### PR TITLE
Tweak OIDC configuration in API and dashboard

### PR DESCRIPTION
**Pull request by jraddaoui**
- Allow to skip API auth. email verified check (API).
- Allow OIDC scopes configuration (dashboard).
- Add OIDC ABAC roles mapping type guard (dashboard).

Refs #1037.

### CODE REVIEW

**Comment by jraddaoui** [comment]
@djjuhasz it's probably better that you review this PR commit by commit.

File: docs/src/admin-manual/iac.md [hunk 1]
```diff
@@ -112,14 +112,13 @@ URLs, to set them in the OIDC provider for this client, they will be:
...
-default. If needed, `VITE_OIDC_EXTRA_SCOPES` can be used to request additional
-scopes separated by whitespace. Similarly, `VITE_OIDC_EXTRA_QUERY_PARAMS` can
-be set to specify further query string parameters to be including in the
-authorization request. E.g, when using Azure AD a resource parameter is
-required, or using Auth0 you may need to send an audience client ID. The
-expected format is key value pairs separated by `=` (`audience=cient-id`), if
-more than one parameter is needed they can be added separated by comma
-(`audience=cient-id,key=value`).
+default. If needed, `VITE_OIDC_SCOPES` can be used to replace those scopes.
+Similarly, `VITE_OIDC_EXTRA_QUERY_PARAMS` can be set to specify further query
```
Comment (djjuhasz) [inline docs/src/admin-manual/iac.md]: Remove "Similarly, " as `VITE_OIDC_EXTRA_QUERY_PARAMS` isn't similar to `VITE_OIDC_SCOPES`. ;)

File: docs/src/admin-manual/iac.md [hunk 2]
```diff
@@ -112,14 +112,13 @@ URLs, to set them in the OIDC provider for this client, they will be:
...
+string parameters to be including in the authorization request. E.g, when using
+Azure AD a resource parameter is required, or using Auth0 you may need to send
+an audience client ID. The expected format is key value pairs separated by `=`
+(`audience=cient-id`), if more than one parameter is needed they can be added
```
Comment (djjuhasz) [inline docs/src/admin-manual/iac.md]: typo: "client-id"

File: docs/src/admin-manual/iac.md [hunk 3]
```diff
@@ -112,14 +112,13 @@ URLs, to set them in the OIDC provider for this client, they will be:
...
+separated by comma (`audience=cient-id,key=value`).
```
Comment (djjuhasz) [inline docs/src/admin-manual/iac.md]: typo: "client-id"

File: docs/src/admin-manual/iac.md [hunk 4]
```diff
@@ -112,14 +112,13 @@ URLs, to set them in the OIDC provider for this client, they will be:
...
-authorization request. E.g, when using Azure AD a resource parameter is
-required, or using Auth0 you may need to send an audience client ID. The
-expected format is key value pairs separated by `=` (`audience=cient-id`), if
-more than one parameter is needed they can be added separated by comma
-(`audience=cient-id,key=value`).
+default. If needed, `VITE_OIDC_SCOPES` can be used to replace those scopes.
+Similarly, `VITE_OIDC_EXTRA_QUERY_PARAMS` can be set to specify further query
+string parameters to be including in the authorization request. E.g, when using
+Azure AD a resource parameter is required, or using Auth0 you may need to send
+an audience client ID. The expected format is key value pairs separated by `=`
```
Comment (djjuhasz) [inline docs/src/admin-manual/iac.md]: I think "key-value" should be hyphenated.

**Review by djjuhasz — APPROVED** [review]
Nice @jraddaoui! :+1: I just flagged a few minor grammar and spelling mistakes.

**Pull request merged**